### PR TITLE
Update nom to version 6

### PR DESCRIPTION
**Pull request by ekse**

### CODE REVIEW

File: saphir/src/multipart/parser.rs [hunk 1]
```diff
@@ -313,7 +314,7 @@ mod tests {
         let boundary_partial = &b"------AaB"[..];
         let empty = &b""[..];
 
-        assert_eq!(tag_boundary(boundary)(boundary_partial), Err(nom::Err::Incomplete(Needed::Size(10))));
+        assert_eq!(tag_boundary(boundary)(boundary_partial), Err(nom::Err::Incomplete(Needed::Size(NonZeroUsize::new(3).unwrap()))));
```
Comment (ekse) [inline saphir/src/multipart/parser.rs:317]: The "breaking changes" section for the 6.0 release (https://github.com/Geal/nom/blob/master/CHANGELOG.md#breaking-changes) mentions that "streaming parsers return the number of additional bytes they need, not the total. This was supposed to be the case everywhere, but some parsers were forgotten". Looking at the test, it seems that the correct value should indeed have been 3 and not 10.

**Comment by ekse** [comment]
This replaces pull request #118.

**Comment by richerarc** [comment]
Some lints are failing

**Comment by ekse** [comment]
Ça semble être de nouveaux checks dans clippy, les lignes sont dans des fichiers que je n'ai pas touché.

**Comment by ekse** [comment]
I fixed the clippy lints.

**Review by richerarc — APPROVED** [review]

**Pull request merged**